sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_datastore/one_lake.py

```diff
@@ -0,0 +1,47 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use
+
+from typing import Any, Dict
+
+from marshmallow import fields, post_load
+
+from azure.ai.ml._restclient.v2023_04_01_preview.models import DatastoreType
+
+from azure.ai.ml._schema.core.fields import NestedField, PathAwareSchema, StringTransformedEnum, UnionField
+from azure.ai.ml._utils.utils import camel_to_snake
+
+from .credentials import NoneCredentialsSchema, ServicePrincipalSchema
+
+class OneLakeArtifactSchema(PathAwareSchema):
+    artifact_name = fields.Str(required=True)
+    artifact_type = fields.Str()
+
+class OneLakeSchema(PathAwareSchema):
+    name = fields.Str(required=True)
+    id = fields.Str(dump_only=True)
+    type = StringTransformedEnum(
+        allowed_values=DatastoreType.ONE_LAKE,
+        casing_transform=camel_to_snake,
+        required=True,
+    )
+    # required fields for OneLake
+    one_lake_workspace_name = fields.Str(required=True)
+    artifact = NestedField(OneLakeArtifactSchema)
+    endpoint = fields.Str()
+    # ServicePrincipal and UserIdentity are the two supported credential types
+    credentials = UnionField(
+        [
+            NestedField(ServicePrincipalSchema),
+            NestedField(NoneCredentialsSchema),
```

Comment:
Is NoneCredentialsSchema the same as UserIdentity?

Reply:
Yup, my understanding is that there isn't one for UserIdentity and NoneCredentialsSchema is used to indicate using UserIdentity for the supported cases.